unity/Assets/StreamingAssets/content/MoM/base/images.ini

```diff
@@ -25,24 +25,471 @@ image="{import}/img/CutsceneFrame"
 [ImageCutsceneTextTopper]
 image="{import}/img/CutsceneTextTopper"
 
+[ImageChurch]
+image="{import}/img/Image_1"
```

Comment:
I don't like these names, it looks impossible to maintain.  Can we at least get the source package into the path?

Reply:
I will look it up. Do you mean the section or the value (naming of the file)?

Reply:
The problem: The same image is in different named asset files.

Image = Arkham Campus

asset file                       | asset name        | Platform
---------------------------------|-------------------|-------------
0b21e49278d204bbbb8516e867ac591b | Image_2.tex       | Android
resources.asset                  | Image_2_00001.tex | PC

This is just not tested Pseudocode but we could populate a list of fingerprint for every texture (a new ini files). 
On demand, we could provide the fingerprint instead of the image. In that case the image would be looked up in the file with all the fingerprints.

```
// https://jenssegers.com/61/perceptual-image-hashes
public static string CreateFingerPrint(UnityEngine.Texture2D texture)
{
	// Clone texture and scale it to a 8x8 image
	UnityEngine.Texture2D clone = new UnityEngine.Texture2D(texture.width, texture.height);
	UnityEngine.Graphics.CopyTexture(texture, clone);
	clone.Resize(8, 8, UnityEngine.TextureFormat.ETC2_RGBA8, false);
	clone.Apply();
	// Convert all colors to grey
	UnityEngine.Color[] pixels = clone.GetPixels();
	var fingerPrint = new List<byte>();
	pixels.ToList().ForEach(x => fingerPrint.Add((byte)(pixels[0].grayscale * 255f)));
	// we end up with 64 bytes of fingerprint
	return Convert.ToBase64String(fingerPrint.ToArray());
}

public static bool MatchesFingerprint(string a, string b)
{
	byte[] ab = Convert.FromBase64String(a);
	byte[] bb = Convert.FromBase64String(b);
	if (ab.Length != bb.Length) return false;
	if (ab.Equals(bb)) return true;
	var matches = new List<bool>();
	for (int i = 0; i < ab.Length; i++)
	{
		int allowedDifference = 10;
		matches.Add(Math.Abs(ab[i] - bb[i]) <= allowedDifference); // 10 shades of gray difference allowed here
	}
	return matches.Where(x => x == true).Count() > 5; // allow 5 pixel to be off
}
```